app/controllers/nonsipp/otherassetsdisposal/ReportedOtherAssetsDisposalListController.scala

```diff
@@ -184,15 +192,32 @@ class ReportedOtherAssetsDisposalListController @Inject()(
       )
     }
 
-  def onPreviousViewOnly(srn: Srn, page: Int, year: String, current: Int, previous: Int): Action[AnyContent] =
-    identifyAndRequireData(srn).async {
-      Future.successful(
-        Redirect(
-          controllers.nonsipp.otherassetsdisposal.routes.ReportedOtherAssetsDisposalListController
-            .onPageLoadViewOnly(srn, page, year, (current - 1).max(0), (previous - 1).max(0))
-        )
+  def onPreviousViewOnly(
+    srn: Srn,
+    page: Int,
+    year: String,
+    current: Int,
+    previous: Int
+  ): Action[AnyContent] = identifyAndRequireData(srn, ViewOnlyMode, year, (current - 1).max(0), (previous - 1).max(0)) {
+    implicit request =>
+      val showBackLink = false
+      val viewOnlyViewModel = ViewOnlyViewModel(
+        viewOnlyUpdated = if (ViewOnlyMode == ViewOnlyMode && request.previousUserAnswers.nonEmpty) {
```

Comment:
I think there's a typo here as `ViewOnlyMode == ViewOnlyMode` will always evaluate as true, did you mean `mode == ViewOnlyMode`?

Reply:
Thats correct, It should be Mode==viewOnlyMode
That should help the test case !